### DISCUSSION TITLE
how to handle form reset without validation trigger?

### DISCUSSION THREAD

**Discussion by megacherry** [TanStack/form#984] [Q&A]
I have a form, like so:

```
const form = useForm({
    onSubmit: async ({ value }) => {
      form.reset()
    },
    defaultValues: {
      message: '',
    },
    validatorAdapter: valibotValidator(),
  })
```

when the form is reset by a submit, the validation is triggered again. I found nothing in the docs.

Sample code:

```
<form.Field
        name="message"
        validators={{
          onChange: v.pipe(v.string(), v.minLength(10), v.maxLength(1000)),
        }}
        children={(field) => (
            <Textarea
              placeholder="message"
              value={field.state.value}
              onBlur={field.handleBlur}
              onChange={(e) => field.handleChange(e.nativeEvent.text)}
            />
        )}
      />

      <form.Subscribe
        selector={(state) => [state.canSubmit, state.isSubmitting]}
        children={([canSubmit, isSubmitting]) => (
          <Button
            className="w-full self-center"
            disabled={!canSubmit || isSubmitting}
            onPress={form.handleSubmit}
          >
            {isSubmitting ? (
              <ActivityIndicator />
            ) : (
              <Text className="uppercase">Submit</Text>
            )}
          </Button>
        )}
      />
```

**Comment by crutchcorn** [comment]
This doesn't make sense to me. What usecase would you want to allow the user to submit an invalid form?

**Comment by megacherry** [comment]
I want to present the user a cleared form (message is empty) after submit.

**Comment by wasabii** [comment]
@crutchcorn I can explain pretty well. Just hit this.

I have a complex form that is displayed to both an end user, and then validated by an administrator. The end user should not be able to submit it without validation. However: the admin should be able to skip validation. That is the admin is allowed in this case to override many of the values to make them invalid.

At first I thought I would just use different schema depending on who you are. Except the admin wants to be able to SEE that the form is invalid, and make the decision explicitly to submit anyways.

**Reply by wasabii** [reply]
What would be nice for me is a way to have two schema, actually. One for errors, and another for warnings.
